draft-barnes-acme.md

```diff
@@ -846,7 +856,18 @@ token (required, string):
 
 ~~~~~~~~~~
 
-A client responds to this Challenge by provisioning the token as a resource on the HTTP server for the domain in question.  The path at which the resource is provisioned is determined by the client, but MUST begin with ".well-known/acme-challenge/".  The content type of the resource MUST be "text/plain".  The client returns the part of the path coming after that prefix in its Response message.
+A client responds to this challenge by signing a JWS object and provisioning it as a resource on the HTTP server for the domain in question.  The payload of the JWS MUST be a JSON dictionary containing the fields "type", "token", "path", and "tls" from the ACME challenge and response, and no other fields.  The JWS MUST be signed with the client's account key pair, and MUST meet the guidelines laid out in {{terminology}} above.
+
+~~~~~~~~~~
+{
+  "type": "simpleHttp",
+  "token": "evaGxfADs6pSRb2LAv9IZf17Dt3juxGJ+PCt92wr+oA",
+  "path": "6tbIMBC5Anhl5bOlWT5ZFA",
+  "tls": false,
+}
+~~~~~~~~~~
+
+The path at which the resource is provisioned is determined by the client, but MUST begin with ".well-known/acme-challenge/".  The content type of the resource, if provided, MUST be "application/jose+json".  In addition to expressing the path in the JWS as described above, the client returns the part of the path coming after that prefix in its Response message.
```

Comment:
I like the `text/plain` -> `application/jose+json` change, but I disagree with the `MUST` -> `if provided, MUST`.

One of our big concerns with SimpleHTTPS is that common app configurations may allow people to upload arbitrary files, including to `/.well-known/`. One way to mitigate that risk is to require a specific content-type. So, for instance, if you allow uploading of arbitrary images, but only serve them with content-types appropriate for images, you would not be vulnerable under the old language and you would be vulnerable under the new language.